x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichLookupService.java

```diff
@@ -300,9 +300,12 @@ private void doLookup(
                 new ValuesSourceReaderOperator(
                     driverContext.blockFactory(),
                     fields,
-                    List.of(new ValuesSourceReaderOperator.ShardContext(searchContext.searcher().getIndexReader(), () -> {
-                        throw new UnsupportedOperationException("can't load _source as part of enrich");
-                    })),
+                    List.of(
+                        new ValuesSourceReaderOperator.ShardContext(
+                            searchContext.searcher().getIndexReader(),
+                            searchContext::newSourceLoader
```

Comment:
I remember sometime ago we talked about `_source` not being available in an enrich index. But it sure is there in the test case I wrote.

Reply:
This brings the topic of returning _source as a  column. 
Let's turn off returning _source by default and look into opening it up in the future, similar to the approach we have for FROM.

Reply:
We don't return it by default here either - this just lets the text fields use it if they need it.

Reply:
We explicitly enable _source for enrich indices: https://github.com/elastic/elasticsearch/blob/b7eafce32cb1e7dfdcb259aab89b886ff93a1c66/x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunner.java#L383

Reply:
Ah! I thought it was the other way around. Either way, that's fine.

I suppose it'd be nice if we told people to prefer non-text fields for enrich. They aren't going to be fast.

Reply:
I think we can fix this by using a keyword instead of non-indexed text when executing an enrich policy. This change should be backward compatible since the ingest enrich process use only _source; and ESQL works with either way. I will open a PR for this.

Reply:
Sure. The `ignore_above` option of a keyword can be an issue here though. I will do more testing.

Reply:
Yeah. That's annoying because it won't matter unless the string is > 32kb, right? If we just don't turn on the option it won't get in the way. But if the fields are >32kb it'll break the doc values.